lasagne/layers/cuda_convnet.py

```diff
@@ -244,24 +241,20 @@ def __init__(self, incoming, num_filters, filter_size, stride=(1, 1),
                                "channels to be 1, 2, 3 or a multiple of 4, "
                                "but it is %d" % self.num_input_channels)
 
-        if border_mode is not None and pad is not None:
-            raise RuntimeError("You cannot specify both 'border_mode' and "
-                               "'pad'. To avoid ambiguity, please specify "
-                               "only one of them.")
-        elif border_mode is None and pad is None:
+        if pad is None:
             # no option specified, default to valid mode
             self.pad = 0
-        elif border_mode is not None:
-            if border_mode == 'valid':
+        elif isinstance(pad, basestring):
+            if pad == 'valid':
                 self.pad = 0
-            elif border_mode == 'full':
+            elif pad == 'full':
                 self.pad = self.filter_size - 1
-            elif border_mode == 'same':
+            elif pad == 'same':
                 # only works for odd filter size, but the even filter size case
                 # is probably not worth supporting.
                 self.pad = (self.filter_size - 1) // 2
             else:
-                raise RuntimeError("Invalid border mode: '%s'" % border_mode)
+                raise RuntimeError("Invalid pad: '%s'" % pad)
```

Comment:
I'd change the if/then/else construct a bit as suggested above, but maybe the `basestring` check is not that bad anyway. This time, `'full'` and `'same'` cannot be handled by the implementation and need to be converted to explicit padding as done here :+1: